src/Microsoft.Diagnostics.NETCore.Client/ReversedServer/ReversedDiagnosticsServer.cs

```diff
@@ -0,0 +1,425 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    /// <summary>
+    /// Establishes server endpoint for runtime instances to connect when
+    /// configured to provide diagnostic endpoints in reverse mode.
+    /// </summary>
+    internal sealed class ReversedDiagnosticsServer : IDisposable
```

Comment:
To match the new terminology that we are discussing, I'm proposing to rename this to DiagnosticPortListener.

Reply:
Holding off on this until new terminology is settled.